### PR TITLE
Feature: Added read_config + validate_config for JSON harmonization configuration

### PR DESCRIPTION
**Pull request by Naalu**
Adds two new functions for reading and validating the JSON harmonization config that drives all of the downstream data transformations.

## New functions

- `read_config(config.path)` — reads `variable-config.json` using `RJSONIO::fromJSON()`, returns the parsed R list with original structure preserved (`$desired_variables` + `$transformations`).
- `validate_config(config, do.list = NULL)` — checks structural integrity: non-empty `desired_variables`, equal-length transform vectors, required fields in rename/merge entries. Optionally cross-references against `parse_do()` metadata and warns for missing variables.

## New files

- `R/read_config.R`, `R/validate_config.R` — implementations
- `tests/testthat/test-read_config.R` — 5 tests (structure, types, required fields, missing file, malformed JSON)
- `tests/testthat/test-validate_config.R` — 8 tests (valid config, empty/non-char desired_variables, mismatched lengths, missing fields, do.list cross-ref, bundled config)
- `man/read_config.Rd`, `man/validate_config.Rd` — the associated man pages
- `inst/extdata/variable-config.json` — *Important:* This is just a copy of the bundled config from the prep repo, we will still need to update the config for 2024

## Dependencies

- Adds `RJSONIO` to `Imports` in DESCRIPTION and `importFrom(RJSONIO, fromJSON)` in NAMESPACE.

## Notes

- Based on `read_dta` branch (PR#5).
- `validate_config` tests use synthetic minimal configs for edge cases, following the pattern from `test-read_dta.R`.

### CODE REVIEW

File: man/validate_config.Rd [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    \item Each \code{$transformations$transform} entry has
+      equal-length \code{years}, \code{value}, \code{new_value}, and
+      \code{new_label} vectors.
+    \item Each \code{$transformations$rename_columns} entry has
+      \code{years} and \code{new_name}.
+    \item Each \code{$transformations$merge_columns} entry has
+      \code{years}, \code{column_1}, and \code{column_2}.
+  }
+
+  If \code{do.list} is provided, additionally warns for any desired
```
Comment (tdhock) [inline man/validate_config.Rd]: it seems there are two kinds of issues.
one kind results in warnings (documented here).
the other kind results in an error.
I would suggest a different design, maybe with an intermediate function `config_issues()` that returns a table with one row for each issue detected. (zero rows for no issues)
Or is that overkill?
What do you think?

**Review by tdhock — COMMENTED** [review]
looks good overall, but please consider and respond to questions.

File: man/validate_config.Rd [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  \enumerate{
+    \item \code{desired_variables} is a non-empty character vector.
+    \item Each \code{$transformations$transform} entry has
+      equal-length \code{years}, \code{value}, \code{new_value}, and
+      \code{new_label} vectors.
+    \item Each \code{$transformations$rename_columns} entry has
+      \code{years} and \code{new_name}.
+    \item Each \code{$transformations$merge_columns} entry has
+      \code{years}, \code{column_1}, and \code{column_2}.
```
[the remark this answers is not in the document]
Comment (Naalu) [reply man/validate_config.Rd]: Good point — `column_1`/`column_2` are the field names from the existing `variable-config.json` (from the prep repo), not R variable names I chose. The merge works as a coalesce: prefer `column_1` where non-NA, and fall back to `column_2`. Neither is really 'lost' — both contribute values based on missingness. I agree that descriptive names are better. How about `column_primary` and `column_secondary`? That reflects the priority/fallback relationship. However, this would require updating the JSON config file — happy to do that in a follow-up commit if you like those names (or suggest alternatives).

[the remark this answers is not in the document]
Comment (tdhock) [reply man/validate_config.Rd]: thanks.
primary/secondary is just another way of writing 1 2 so I would avoid that too.
preferred/fallback?

File: inst/extdata/variable-config.json [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+				"new_label": ["Other: Relative", "There is only one primary adult caregiver in the household for this child"]
+			},
+			"family": {
+				"years": ["2016"],
+				"value": ["1", "2", "3", "4", "5", "6", "7", "8", "9"],
+				"new_value": ["1", "2", "3", "4", "5", "5", "8", "8", "8"],
+				"new_label": ["Two biogical/adoptive parents, currently married", "Two biogical/adoptive parents, not currently married", "Two parents (at least one not biological/adoptive), currently married", "Two parents (at least one not biological/adoptive), not currently married", "Single mother", "Single mother", "Other relation", "Other relation", "Other relation"]
+			},
+			"hcability": {
+				"years": ["2016", "2017", "2018"],
```
[the remark this answers is not in the document]
Comment (tdhock) [reply inst/extdata/variable-config.json]: I’m not sure to understand.
If they are sometimes needed as strings in the code, ok to keep as strings in the json.
if they are used as numbers in the R code, then we could simplify both the code (remove conversion) and the json (remove quotes).
simpler is better because faster and easier to understand.

[the remark this answers is not in the document]
Comment (Naalu) [reply inst/extdata/variable-config.json]: Totally understand where you're coming from, but I think this is the simpler approach (even though I don't entirely love it). Years look like numbers, but they're identifiers — we never do arithmetic on them. They're compared with `as.character(year) %in% details$years` throughout the previous codebase. If we unquote years in the JSON, `fromJSON()` returns them as floating-point doubles, and we'd need something like `year %in% as.integer(details$years)` everywhere instead, which I don't think is particularly simpler and is probably more error-prone to make the switch, given the previous code.

The other benefit, in my opinion, is consistency: this approach ensures uniform types across all fields in each config entry — every array is character (this was the previous config's approach).

The as.numeric() conversion is one line per use and, though a little clunky, makes the intent explicit. A mixed approach (some fields quoted, some not) seems more error-prone for anyone editing the JSON later.

File: man/validate_config.Rd (hunk 2 above)
[the remark this answers is not in the document]
Comment (Naalu) [reply man/validate_config.Rd]: Addressed in PR #14

File: inst/extdata/variable-config.json [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+			"a2_relation": {
+				"years": ["2016", "2017", "2018"],
+				"value": ["5", "8"],
+				"new_value": ["6", "8"],
+				"new_label": ["Other: Relative", "There is only one primary adult caregiver in the household for this child"]
+			},
+			"family": {
+				"years": ["2016"],
+				"value": ["1", "2", "3", "4", "5", "6", "7", "8", "9"],
+				"new_value": ["1", "2", "3", "4", "5", "5", "8", "8", "8"],
```
Comment (tdhock) [inline inst/extdata/variable-config.json]: here also, I would suggest removing quotes if possible

File: R/transform_values.R [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  for(variable.name in names(transforms)){
+    details <- transforms[[variable.name]]
+    if(as.character(year) %in% details$years &&
+       variable.name %in% names(dt)){
+      label.col <- paste0(variable.name, "_label")
+      ## Initialize _label column if it does not yet exist.
+      if(!(label.col %in% names(dt))){
+        data.table::set(dt, j = label.col, value = NA_character_)
+      }
+      for(i in seq_along(details$value)){
```
[the remark this answers is not in the document]
Comment (tdhock) [reply R/transform_values.R]: #22

**Review by tdhock — APPROVED** [review]
